### PR TITLE
Fix periodic free surfaces in spherical shell geometries

### PR DESCRIPTION
**Pull request by Minerallo**
Hi everyone,
While running some quick tests, I realised that mesh deformation in a spherical shell with periodic boundary conditions was not working correctly and crashing.
The mesh-deformation now asks the geometry model to create the periodic constraints, which applies the rotation for the spherical shell.
Cheers,
Michael

### For all pull requests:

* [x] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).
* I followed the [AI Policy](../blob/main/CONTRIBUTING.md#policy-on-usage-of-ai):
  - [x] Significant parts of this PR were written by AI (please add a sentence below).
  - [ ] AI has not been used in significant ways.

*If yes, please describe your usage of AI models in the creation of this pull request*

### For new features/models or changes of existing features:

* [x] I have tested my new feature locally to ensure it is correct.
* [x] I have [created a testcase](https://aspect-documentation.readthedocs.io/en/latest/user/extending/testing/writing-tests.html) for the new feature/benchmark in the [tests/](../blob/main/tests/) directory.
* [x] I have added a changelog entry in the [doc/modules/changes](../blob/main/doc/modules/changes) directory that will inform other users of my change.

### CODE REVIEW

File: source/mesh_deformation/interface.cc [hunk 1]
```diff
@@ -789,21 +784,18 @@ namespace aspect
...
-      // Add the vanilla periodic boundary constraints
+      // Record the periodic boundary ids and let the geometry model join
+      // their values, including any rotation needed by curved geometries.
       std::set<types::boundary_id> periodic_boundaries;
       using periodic_boundary_pairs = std::set<std::pair<std::pair<types::boundary_id, types::boundary_id>, unsigned int>>;
       const periodic_boundary_pairs pbp = this->get_geometry_model().get_periodic_boundary_pairs();
       for (const auto &p : pbp)
         {
           periodic_boundaries.insert(p.first.first);
           periodic_boundaries.insert(p.first.second);
```
Comment (bangerth) [inline source/mesh_deformation/interface.cc]: You no longer need to build `periodic_boundaries`, the variable isn't used below any more after your changes.

File: tests/periodic_quarter_shell_free_surface.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+"$(dirname "$0")/cmake/default" "$1" |
+  grep -v -E "shmem: mmap: an error occurred|btl_tcp_component_create_listen.*bind\\(\\) failed|^-+$|^\\+[-+]+\\+$" |
+  perl -0pe 's/\n+\z/\n/'
```
Comment (bangerth) [inline tests/periodic_quarter_shell_free_surface.sh]: I must admit that I don't understand what this does. Would you mind prefixing it with a comment that explains what the script tries to achieve?

**Review by bangerth — COMMENTED** [review]
Yes, nice find!

File: tests/periodic_quarter_shell_free_surface.sh (hunk 2 above)
Comment (tjhei) [reply tests/periodic_quarter_shell_free_surface.sh]: I think this .sh file needs to be removed. It points at some MPI misconfiguration on your machine, @Minerallo

**Review by gassmoeller — COMMENTED** [review]
Very nice! Please add a changelog entry that describes the bug you fixed. Then this is ready to merge.

**Comment by Minerallo** [comment]
> Very nice! Please add a changelog entry that describes the bug you fixed. Then this is ready to merge.

thanks all done !

**Review by gassmoeller — APPROVED** [review]

**Comment by bangerth** [comment]
Some of the quicktests are failing. Is that expected?

File: tests/periodic_quarter_shell_free_surface.prm [hunk 3]
```diff
@@ -0,0 +1,96 @@
+# Enable if: QUICK_TEST
```
Comment (gassmoeller) [inline tests/periodic_quarter_shell_free_surface.prm]: this is the reason the test is failing. please remove the tag as QUICK_TEST, otherwise the MacOS tester will run it as well, and that one produces different results.
